deepface/DeepFace.py

```diff
@@ -702,23 +716,23 @@ def represent(
             align=align,
         )
     else:  # skip
-        if isinstance(img_path, str):
-            img = functions.load_image(img_path)
-        elif type(img_path).__module__ == np.__name__:
+        if type(img_path).__module__ == np.__name__:
             img = img_path.copy()
         else:
-            raise ValueError(f"unexpected type for img_path - {type(img_path)}")
+            # Try load. If load error, will raise exception internal
+            img, _ = functions.load_image(img_path)
         # --------------------------------
         if len(img.shape) == 4:
             img = img[0]  # e.g. (1, 224, 224, 3) to (224, 224, 3)
         if len(img.shape) == 3:
             img = cv2.resize(img, target_size)
-            img = np.expand_dims(img, axis=0)
-            # when represent is called from verify, this is already normalized
+            img = np.expand_dims(img, axis=0) # Why we remove a axis=0 previously and here expand one?
```

Comment:
Here need author review. I don't know why should this, so I add a comment without change anything.

Reply:
face recognition models expect inputs like (1, 224, 224, 3). so, this is intended behaviour. please remove your comment.

Reply:
but Why we do `img = img[0]` in 726

Reply:
it is expanding. if your input is (224, 224, 3) then this makes (1, 224, 224, 3). your command drops a dimension on the other hand.